Bottom: c50899f57171b8e99a6c9eec28bfb6bbae89565c
Top:    5802825b21db97475d4bb81a96311426062d7e04
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-30 11:37:35 +0000

Refresh of allow-polygons-with-clearlinef

---

diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index 2d858be..507a5fb 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -264,7 +264,7 @@ common_gui_draw_pcb_polygon (hidGC gc, PolygonType *polygon, const BoxType *clip
 
   if (TEST_FLAG (THINDRAWFLAG,     PCB) ||
       TEST_FLAG (THINDRAWPOLYFLAG, PCB) ||
-      TEST_FLAG (CLEARLINEFLAG,    PCB))
+      TEST_FLAG (CLEARLINEFLAG, polygon))
     gui->graphics->thindraw_pcb_polygon (gc, polygon, clip_box);
   else
     gui->graphics->fill_pcb_polygon (gc, polygon, clip_box);
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index e466999..57f817a 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -691,6 +691,11 @@ ghid_thindraw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
 {
   double old_alpha_mult = gc->alpha_mult;
   common_thindraw_pcb_polygon (gc, poly, clip_box);
+
+  /* Don't shade anti-polygons */
+  if TEST_FLAG (CLEARLINEFLAG, poly)
+    return;
+
   ghid_set_alpha_mult (gc, gc->alpha_mult * 0.25);
   gui->graphics->fill_pcb_polygon (gc, poly, clip_box);
   ghid_set_alpha_mult (gc, old_alpha_mult);
